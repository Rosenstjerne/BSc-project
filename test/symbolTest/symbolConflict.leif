class x {
    var int y;
}
var int x;

function int x() {
    return 42;
}

x = 5;


# This test should raise an error, as 'x' is used as a variable, class, and function name