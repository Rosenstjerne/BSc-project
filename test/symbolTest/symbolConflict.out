Traceback (most recent call last):
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\compiler.py", line 174, in <module>
    main(sys.argv[1:])
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\compiler.py", line 164, in main
    result = compiler(show_source, show_ast, macOS, input_file, output_file)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\compiler.py", line 86, in compiler
    the_program.accept(symbols_collector)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\AST.py", line 140, in accept
    self.body.accept(visitor)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\AST.py", line 22, in accept
    self.class_decl.accept(visitor)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\AST.py", line 53, in accept
    self.decl.accept(visitor)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\AST.py", line 74, in accept
    visitor.postVisit(self)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\visitors_base.py", line 28, in postVisit
    self._visit(t, "postVisit")
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\visitors_base.py", line 13, in _visit
    method(t)
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\symbols.py", line 273, in postVisit_class_declaration
    values = {(a.name):(SymVal(NameCategory.VARIABLE,self._current_level,a,a.vtype)) for a in t.attLst} 
  File "C:\Users\andr9\OneDrive\Dokumenter\BSc-project\src\symbols.py", line 273, in <dictcomp>
    values = {(a.name):(SymVal(NameCategory.VARIABLE,self._current_level,a,a.vtype)) for a in t.attLst} 
AttributeError: 'variables_list' object has no attribute 'name'
